### PR TITLE
Migrate LGPO tests to pytest

### LINKED ISSUE #61717: [TECH DEBT] Fix lgpo module integration tests
### Description of the tech debt to be addressed, include links and screenshots
The `tests.integration.modules.test_win_lgpo` tests aren't even running. They need the whitelist decorator added and the lgpo.exe setup fixed. Then some of the tests themselves are failing.

### Versions Report
3004

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Migrates LGPO tests to pytest. Enables the tests so they run.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/61717

### Previous Behavior
LGPO tests were not running

### New Behavior
LGPO tests now run

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by whytewolf — DISMISSED** [review]

**Comment by whytewolf** [comment]
@twangboy pre-commit fail.

**Review by whytewolf — DISMISSED** [review]

File: tests/pytests/functional/modules/win_lgpo/test_admx_policy.py [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+
+@pytest.fixture(scope="module")
+def osrelease():
+    osrelease_grains = grains.item("osrelease")
+    yield osrelease_grains.get("osrelease", None)
+
+
+@pytest.fixture(scope="module")
+def lgpo_bin():
+    if not os.path.exists(r"C:\Windows\System32\lgpo.exe"):
```
Comment (Ch3LL) [inline tests/pytests/functional/modules/win_lgpo/test_admx_policy.py:56]:
```suggestion
    sys_dir = pathlib.Path("C:\\", "Windows", "System32", "lgpo.exe")
    if not (sys_dir / "lgpo.exe").exists():
```

File: tests/pytests/functional/modules/win_lgpo/test_admx_policy.py [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+        sys_dir = pathlib.Path(r"C:\Windows\System32")
```
Comment (Ch3LL) [inline tests/pytests/functional/modules/win_lgpo/test_admx_policy.py:57]:
```suggestion

```

**Review by whytewolf — DISMISSED** [review]

File: tests/pytests/functional/modules/win_lgpo/test_admx_policy.py [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+                    location = z.extract(file, path=r"C:\Windows\System32")
+        # Move lgpo.exe to system32
+        log.debug("Placing LGPO.exe in System32")
+        lgpo_bin = pathlib.Path(location)
+        lgpo_bin = lgpo_bin.rename(sys_dir / lgpo_bin.name.lower())
+        yield lgpo_bin
+        log.debug("Cleaning up LGPO artifacts")
+        zip_file.unlink(missing_ok=True)
+        lgpo_bin.unlink(missing_ok=True)
+        if os.path.exists(sys_dir / "LGPO_30"):
```
Comment (Ch3LL) [inline tests/pytests/functional/modules/win_lgpo/test_admx_policy.py:79]:
```suggestion
        if (sys_dir / "LGPO_30").exists():
```

File: tests/pytests/functional/modules/win_lgpo/test_admx_policy.py [hunk 4]
```diff
@@ -0,0 +1,699 @@
...
+    reg_pol = pathlib.Path(r"C:\Windows\System32\GroupPolicy\User\Registry.pol")
+    reg_pol.unlink(missing_ok=True)
+    _test_set_user_policy(
+        lgpo_bin=lgpo_bin,
+        shell=shell,
+        name=name,
+        setting=setting,
+        exp_regexes=exp_regexes,
+    )
+    reg_pol.unlink()
```
Comment (Ch3LL) [inline tests/pytests/functional/modules/win_lgpo/test_admx_policy.py:480]: same here

**Review by dmurphy18 — DISMISSED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by whytewolf — APPROVED** [review]

**Pull request merged**